neoq.go

```diff
@@ -255,6 +251,19 @@ func ConnectionString(connectionString string) ConfigOption {
 	}
 }
 
+// AddLogger configures neoq to use the spcified logger
+func AddLogger(logger Logger) ConfigOption {
+	return func(n Neoq) {
+		switch b := n.(type) {
+		case *MemBackend:
```

Comment:
We can get around this by forcing a set logger function in the interface.

Reply:
I believe the same argument can be made for any `ConfigOption` function, and I'd like to keep the `Neoq` interface minimal. 

Perhaps there's an argument to be made that setting a logger is an elevated configuration option, deserving of its own place in the `Neoq` interface.

Reply:
I think I agree with that. Logging should be a first class citizen. Alas it isn't in a lot of things haha.

Reply:
Cool, I'm hip to that.

Reply:
Logging should indeed be first class. I am quite torn though. I think logging should be first class _and_ interfaces should be as small as possible. 

At the moment, those two things are in contention with on another. 

I feel more inclined to grant logging its first class status with a dedicated `NewWithLogger` than by adding it to the `Neoq` interface. `pgx/pgxpool` has a similar pattern: https://pkg.go.dev/github.com/jackc/pgx/v5/pgxpool#NewWithConfig

What do you think?

Reply:
If it's not part of the interface then you have to do the check for all types of types and set the logger differently on each.

having it part of the interface makes it a common thing so you call 'SetLogger' on any

Reply:
Done